### PR TITLE
minor optimizations

### PR DESCRIPTION
**Pull request by GopherJ**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -550,7 +550,7 @@ where
 }
 
 /// The error type returned by methods in this crate.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
```
Comment (GopherJ) [inline src/lib.rs:553]: this is needed for some applications which has its own Error and it wants to derive PartialEq

File: Cargo.toml [hunk 2]
```diff
@@ -10,6 +10,6 @@ keywords = ["database", "pool"]
 edition = "2018"
 
 [dependencies]
-log = "0.4"
-parking_lot = "0.12"
-scheduled-thread-pool = "0.2"
+log = "^0.4"
+parking_lot = "^0.12"
+scheduled-thread-pool = "^0.2"
```
Comment (Mingun) [inline Cargo.toml:15]: Cargo by default [treats all dependencies as caret dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements), so this does not change anything

Comment (sfackler) [reply Cargo.toml:15]: It doesn't change anything, so why is it part of the PR?

Comment (GopherJ) [reply Cargo.toml:15]: removed, thanks for letting me know

**Comment by sfackler** [comment]
See #141 for why Error doesn't implement PartialEq.

**Pull request closed without merging**